plugins/modules/fabric_virtual_networks_workflow_manager.py

```diff
@@ -0,0 +1,2950 @@
+        """
+        Checks if the Anycast Gateway configuration needs to be updated based on provided parameters.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            anycast (dict): A dictionary containing the current configuration parameters for the Anycast Gateway.
+            anycast_details_in_ccc (dict): A dictionary containing the current Anycast Gateway configuration details
+                                            from the Cisco Catalyst Center (CCC) to compare against.
+        Returns:
+            bool: True if any of the relevant parameters differ between the provided `anycast` configuration and the
+                corresponding parameters in `anycast_details_in_ccc`, indicating that an update is needed.
+                Returns False if no differences are found.
+        Description:
+            This function compares specific configuration parameters of an Anycast Gateway between the provided
+            configuration (`anycast`) and the existing configuration in the Cisco Catalyst Center.
+            If any differences are detected, it logs an informational message and returns True. If no discrepancies
+            are found, the function returns False, indicating that the current configuration is up to date.
+        """
+
+        update_param_to_check = ["tcp_mss_adjustment", "traffic_type", "security_group_name", "layer2_flooding_enabled", "fabric_enabled_wireless"
+                                 , "ip_directed_broadcast", "multiple_ip_to_mac_addresses", "supplicant_based_extended_node_onboarding"
+                                 , "group_policy_enforcement_enabled"]
+        vn_name = anycast.get("vn_name")
+        anycast_mapping = self.get_anycast_gateway_mapping(vn_name)
+
+        if vn_name == "INFRA_VN":
+            params_to_remove = ["security_group_name", "layer2_flooding_enabled", "fabric_enabled_wireless",
+                                "ip_directed_broadcast", "multiple_ip_to_mac_addresses"]
+            for param in params_to_remove:
+                update_param_to_check.remove(param)
+        else:
+            update_param_to_check.remove("supplicant_based_extended_node_onboarding")
+            update_param_to_check.remove("group_policy_enforcement_enabled")
+
+        if anycast.get("traffic_type") and anycast_details_in_ccc.get("isCriticalPool") is True:
+            update_param_to_check.remove("traffic_type")
+
+        for param in update_param_to_check:
+            if anycast.get(param) is not None:
+                key_in_ccc = anycast_mapping.get(param)
+                if anycast.get(param) != anycast_details_in_ccc.get(key_in_ccc):
+                    msg = (
+                        "Given parameter '{0}' for the given anycast doesnot match so gateway needs update."
+                    ).format(param)
+                    self.log(msg, "INFO")
+                    return True
+
```

Comment:
self.log("No discrepancies found; Anycast Gateway configuration is up to date.", "INFO")
    reutrn False